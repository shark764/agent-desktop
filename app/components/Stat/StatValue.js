/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

/**
 *
 * StatValue
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import TimeStat from 'components/TimeStat';

function StatValue(props) {
  let value;
  if (props.stat.isErrored) {
    value = '-';
  } else {
    let percent;
    switch (props.stat.statAggregate) {
      case 'count':
        value = props.stat.results.count;
        break;
      case 'percent':
        percent = props.stat.results.percent;
        value = `${percent}%`;
        break;
      case 'avg':
      case 'max':
      case 'min':
      case 'total':
        value = (
          <TimeStat
            time={props.stat.results[props.stat.statAggregate]}
            unit="millis"
          />
        );
        break;
      default:
        console.warn(
          '[Agent Desktop] Agent statistic has unknown aggregate key.'
        );
        value = '-';
    }
  }
  return <span>{value}</span>;
}

StatValue.propTypes = {
  stat: PropTypes.object.isRequired,
};

export default StatValue;
