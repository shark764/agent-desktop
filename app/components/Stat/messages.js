/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

/*
 * Stat Messages
 *
 * This contains all the text for the Stat component.
 */
import { defineMessages } from 'react-intl';

export default defineMessages({
  source: {
    id: 'app.components.Stat.source',
    defaultMessage: 'Source',
  },
  statistic: {
    id: 'app.components.Stat.statistic',
    defaultMessage: 'Statistic',
  },
  aggregate: {
    id: 'app.components.Stat.aggregate',
    defaultMessage: 'Aggregate',
  },
  disabledQueue: {
    id: 'app.components.Stat.disabledQueue',
    defaultMessage: '(Queue disabled, name not available)',
  },
});
