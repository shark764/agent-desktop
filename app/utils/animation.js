/**
 * throttle max function call rate to browser refresh rate
 */

const requestAnimationFrame =
  window.requestAnimationFrame ||
  window.webkitRequestAnimationFrame ||
  window.mozRequestAnimationFrame ||
  ((callback) => window.setTimeout(callback, 1000 / 60));

function throttler(...callbackArgs) {
  if (!this.running) {
    this.running = true;
    requestAnimationFrame(() => {
      this.callback(...callbackArgs);
      this.running = false;
    });
  }
}

export function throttleDecorator(funcToThrottle) {
  return throttler.bind({ callback: funcToThrottle, running: false });
}
