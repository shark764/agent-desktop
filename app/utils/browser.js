/*
 * Copyright Â© 2015-2018 Serenova, LLC. All rights reserved.
 */

// Check if browser used is IE11 or greater
// Returns undefined for Chrome|Firefox|Edge
export const isIeEleven = () => document.documentMode && document.documentMode >= 11;
