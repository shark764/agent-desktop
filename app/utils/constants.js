/*
 * Copyright Â© 2015-2020 Serenova, LLC. All rights reserved.
 */

export const colors = {
  lightGray: '#ebebeb',
  offWhite: '#e4e4e4',
  lightRed: '#FE4565',
  regularRed: '#E43D5A',
  darkRed: '#CB3750',
  darkGreen: '#001b1e',
  hoverDarkGreen: 'rgba(0, 27, 30, 0.85)',
  black: '#000',
  white: '#FFFFFF',
  skylightBlue: '#23cdf4',
  toggleChecked: '#54B84F',
  toggleCheckedDisabled: '#b7e3b3',
};
