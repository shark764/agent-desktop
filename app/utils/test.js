import { IntlProvider } from 'react-intl';
import enUS from 'translations/en-US.json';

export const getIntlContext = () => {
  // Construct a new `IntlProvider` instance by passing `props` and
  // `context` as React would, then call `getChildContext()` to get the
  // React Intl API, complete with the `format*()` functions.
  const intlProvider = new IntlProvider({ locale: 'en-US', messages: enUS }, {});
  const { intl } = intlProvider.getChildContext();
  return intl;
};

export default { getIntlContext };
