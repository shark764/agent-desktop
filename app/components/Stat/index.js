/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

/**
*
* Stat
*
*/

import React from 'react';
import { FormattedMessage } from 'react-intl';
import PropTypes from 'prop-types';
import Radium from 'radium';

import IconSVG from 'components/IconSVG';
import StatValue from './StatValue';
import messages from './messages';

function Stat(props) {
  const styles = {
    statBox: {
      width: '104px',
      height: '54px',
      paddingTop: '13px',
      ':hover': {
        boxShadow: 'inset 0 0 6px 1px rgba(0,0,0,0.34)',
      },
    },
    statValue: {
      height: '19px',
      lineHeight: '19px',
      textAlign: 'center',
      color: '#FFFFFF',
      fontWeight: 'bold',
      fontSize: '16px',
    },
    statName: {
      width: '100%',
      padding: '0 4px',
      height: '10px',
      lineHeight: '10px',
      fontSize: '12px',
      fontWeight: '600',
      color: '#FFFFFF',
      textAlign: 'center',
      overflow: 'hidden',
      textOverflow: 'ellipsis',
      whiteSpace: 'nowrap',
    },
    loading: {
      display: 'flex',
      justifyContent: 'center',
    },
    loadingIcon: {
      position: 'relative',
      top: '-5px',
      height: '15px',
    },
    hoverBox: {
      position: 'absolute',
      bottom: '66px',
      right: `${((props.index - 1) * 104) + 19}px`,
      zIndex: '2',
      width: '166px',
      borderRadius: '8px',
      padding: '10px 13px',
      backgroundColor: '#FFFFFF',
      boxShadow: '0 0 6px 1px rgba(0,0,0,0.29)',
    },
    hoverHeader: {
      fontSize: '14px',
      lineHeight: '20px',
      color: '#979797',
    },
    stat: {
      fontSize: '14px',
      lineHeight: '20px',
    },
    statRemove: {
      position: 'absolute',
      right: `${((props.index - 1) * 104) + 60}px`,
      bottom: '30px',
      fontSize: '12px',
      zIndex: '2',
      cursor: 'pointer',
      color: props.readyState === 'notready' ? '#000000' : '#FFFFFF',
      ':hover': {
        color: '#ADD8E6',
      },
    },
    hoverBoxTriangle: {
      position: 'absolute',
      width: '0px',
      height: '0px',
      right: `${((props.index - 1) * 104) + 95}px`,
      bottom: '60px',
      borderWidth: '8px',
      borderStyle: 'solid',
      borderColor: '#FFF transparent transparent #FFF',
      borderImage: 'initial',
      transform: 'rotate(-134deg)',
      borderRadius: '3px',
      boxShadow: '-6px -6px 6px -4px rgba(0,0,0,0.29)',
      zIndex: '2',
    },
  };

  let source;

  switch (props.stat.statSource) {
    case 'resource-id':
      source = 'Agent';
      break;
    case 'queue-id':
      source = props.queues.filter((queue) => queue.id === props.stat.queue)[0].name;
      break;
    case 'tenant-id':
      source = 'Tenant';
      break;
    default:
      console.warn('[Agent Desktop] Agent statistic has unknown source.');
  }

  let aggregate = props.stat.statAggregate;
  if (aggregate === 'avg') {
    aggregate = 'average';
  }
  aggregate = aggregate[0].toUpperCase() + aggregate.slice(1);

  return (
    <div className="stat-box" style={styles.statBox}>
      {props.hover ?
        <span>
          <div style={styles.hoverBox}>
            <div style={styles.hoverHeader}>
              <FormattedMessage {...messages.source} />:
            </div>
            <div style={styles.hoverData}>
              {source}
            </div>
            <div style={styles.hoverHeader}>
              <FormattedMessage {...messages.statistic} />:
            </div>
            <div style={styles.hoverData}>
              {props.userFriendlyName}
            </div>
            <div style={styles.hoverHeader}>
              <FormattedMessage {...messages.aggregate} />:
            </div>
            <div style={styles.hoverData}>
              {aggregate}
            </div>
          </div>
          <span style={styles.hoverBoxTriangle} />
          <span key={props.index} style={styles.statRemove} onClick={() => { props.removeStat(props.stat); }}>X</span>
        </span>
      : ''}
      <div className="stat-value" style={styles.statValue}>
        {(props.stat.results || props.stat.isErrored)
          ? <StatValue stat={props.stat} />
          : <div id="loadingContainer" style={styles.loading}>
            <IconSVG style={styles.loadingIcon} id="loadingIcon" name="loadingWhite" />
          </div>}
      </div>
      <div className="agent-stat" style={styles.statName}>
        {props.userFriendlyName}
      </div>
    </div>
  );
}

Stat.propTypes = {
  userFriendlyName: PropTypes.string.isRequired,
  hover: PropTypes.bool.isRequired,
  index: PropTypes.number.isRequired,
  stat: PropTypes.shape({
    queue: PropTypes.string,
    statAggregate: PropTypes.string.isRequired,
    statSource: PropTypes.string.isRequired,
    results: PropTypes.object,
    isErrored: PropTypes.bool,
  }).isRequired,
  removeStat: PropTypes.func.isRequired,
  readyState: PropTypes.string.isRequired,
  queues: PropTypes.array.isRequired,
};

export default Radium(Stat);
