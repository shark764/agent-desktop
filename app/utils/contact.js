/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

import { isValidEmail, isPossibleNumber } from 'utils/validator';
import isURL from 'validator/lib/isURL';

export const formatValue = (attributeToValidate, value) => {
  let formattedValue;
  switch (attributeToValidate.type) {
    case 'phone':
      formattedValue = value.replace(/[^0-9+*#]/g, '');
      if (formattedValue.indexOf('+') !== 0 && formattedValue.length > 0) {
        formattedValue = `+${formattedValue}`;
      }
      break;
    case 'boolean':
      if (value === 'false' || value === '') formattedValue = false;
      else formattedValue = !!value;
      break;
    default:
      formattedValue = value;
  }
  return formattedValue;
};

export const getError = (attributeToValidate, value) => {
  let error = false;
  if (
    attributeToValidate.mandatory &&
    attributeToValidate.type !== 'boolean' &&
    (value === undefined || value.trim().length < 1)
  ) {
    error = 'errorRequired';
  } else if (value !== undefined && value.length) {
    switch (attributeToValidate.type) {
      case 'email':
        if (!isValidEmail(value)) {
          error = 'errorEmail';
        }
        break;
      case 'phone':
        try {
          if (!isPossibleNumber(value)) {
            error = 'errorPhone';
          }
        } catch (e) {
          error = 'errorPhone';
        }
        break;
      case 'link':
        if (
          !isURL(value, {
            protocols: ['http', 'https'],
            require_protocol: true,
          })
        ) {
          error = 'errorLink';
        }
        break;
      case 'number':
        if (Number.isNaN(Number(value))) {
          error = 'errorNumber';
        }
        break;
      default:
        break;
    }
  }
  return error;
};

export const getLocaleLabel = (attribute, locale) => {
  if (attribute.label[locale] !== undefined) {
    return attribute.label[locale];
  } else if (attribute.label.all !== undefined) {
    // Special case for 'All'. We have our own local translation for it we use.
    return attribute.label.all;
  } else if (attribute.objectName !== undefined) {
    // Fallback for attributes
    return attribute.objectName;
  } else {
    // Fallback for section headers
    return '';
  }
};

export function createSearchQuery(query, nextPage) {
  const encodedQuery = {};
  Object.keys(query).forEach(queryName => {
    const queryToEncode = query[queryName];
    encodedQuery[queryName] = encodeURIComponent(queryToEncode);
  });
  return {
    query: Object.assign(encodedQuery, {
      page: nextPage !== undefined ? nextPage : 1,
    }),
  };
}
