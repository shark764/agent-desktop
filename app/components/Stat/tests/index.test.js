import React from 'react';
import { shallow } from 'enzyme';
import { getIntlContext } from 'utils/test';

import Stat from '../index';

describe('<Stat />', () => {
  it('should render correctly', () => {
    const rendered = shallow(
      <Stat
        id="mockId"
        intl={getIntlContext()}
        statistic={{
          type: 'mockType',
          userFriendlyName: 'mockName',
        }}
        hover
        index={1}
        hoverData={{
          statAggregate: 'mockStatAggregate',
          statSource: 'mockStatSource',
        }}
        toggleStat={() => {}}
        readyState="mockReadyState"
        queues={[]}
      />
    );
    expect(rendered).toMatchSnapshot();
  });
});
