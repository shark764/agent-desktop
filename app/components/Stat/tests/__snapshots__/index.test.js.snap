// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Stat /> should render correctly 1`] = `
<div
  className="stat-box"
  data-radium={true}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  style={
    Object {
      "height": "54px",
      "maxWidth": "104px",
      "minWidth": "104px",
      "paddingTop": "13px",
      "position": "relative",
      "width": "104px",
    }
  }
>
  <div
    className="stat-value"
    data-radium={true}
    style={
      Object {
        "color": "#FFFFFF",
        "fontSize": "16px",
        "fontWeight": "bold",
        "height": "19px",
        "lineHeight": "19px",
        "textAlign": "center",
      }
    }
  >
    <div
      id="loadingContainer"
    >
      <IconSVG
        id="loadingIcon"
        name="loadingWhite"
        style={
          Object {
            "height": "15px",
            "position": "relative",
            "top": "-5px",
            "width": "15px",
          }
        }
      />
    </div>
  </div>
  <div
    className="agent-stat"
    data-radium={true}
    style={
      Object {
        "color": "#FFFFFF",
        "fontSize": "12px",
        "fontWeight": "600",
        "height": "10px",
        "lineHeight": "10px",
        "overflow": "hidden",
        "padding": "0 4px",
        "textAlign": "center",
        "textOverflow": "ellipsis",
        "whiteSpace": "nowrap",
        "width": "100%",
      }
    }
  >
    mockName
  </div>
</div>
`;

exports[`<Stat /> should show hover details on mouse over 1`] = `
<div
  className="stat-box"
  data-radium={true}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  style={
    Object {
      "boxShadow": "inset 0 0 6px 1px rgba(0,0,0,0.34)",
      "height": "54px",
      "maxWidth": "104px",
      "minWidth": "104px",
      "paddingTop": "13px",
      "position": "relative",
      "width": "104px",
    }
  }
>
  <div
    data-radium={true}
    style={
      Object {
        "bottom": "60px",
        "position": "absolute",
        "zIndex": 2,
      }
    }
  >
    <div
      data-radium={true}
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderRadius": "8px",
          "bottom": "-9px",
          "boxShadow": "0 0 6px 1px rgba(0,0,0,0.29)",
          "left": "3px",
          "overflowWrap": "break-word",
          "padding": "10px 13px",
          "position": "relative",
          "width": "130px",
          "wordWrap": "break-word",
          "zIndex": "2",
        }
      }
    >
      <div
        data-radium={true}
        style={
          Object {
            "color": "#979797",
            "fontSize": "14px",
            "lineHeight": "20px",
          }
        }
      >
        <FormattedMessage
          defaultMessage="Source"
          id="app.components.Stat.source"
          values={Object {}}
        />
        :
      </div>
      <div>
        Agent
      </div>
      <div
        data-radium={true}
        style={
          Object {
            "color": "#979797",
            "fontSize": "14px",
            "lineHeight": "20px",
          }
        }
      >
        <FormattedMessage
          defaultMessage="Statistic"
          id="app.components.Stat.statistic"
          values={Object {}}
        />
        :
      </div>
      <div>
        mockName
      </div>
      <div
        data-radium={true}
        style={
          Object {
            "color": "#979797",
            "fontSize": "14px",
            "lineHeight": "20px",
          }
        }
      >
        <FormattedMessage
          defaultMessage="Aggregate"
          id="app.components.Stat.aggregate"
          values={Object {}}
        />
        :
      </div>
      <div>
        MockStatAggregate
      </div>
    </div>
    <div
      data-radium={true}
      style={
        Object {
          "borderColor": "#FFF transparent transparent #FFF",
          "borderImage": "initial",
          "borderRadius": "3px",
          "borderStyle": "solid",
          "borderWidth": "8px",
          "boxShadow": "-6px -6px 6px -4px rgba(0,0,0,0.29)",
          "height": "0px",
          "left": "44px",
          "position": "relative",
          "transform": "rotate(-134deg)",
          "width": "0px",
          "zIndex": "2",
        }
      }
    />
  </div>
  <span
    data-radium={true}
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "color": "#FFFFFF",
        "cursor": "pointer",
        "fontSize": "12px",
        "position": "absolute",
        "right": "10px",
        "top": "6px",
        "zIndex": "2",
      }
    }
  >
    X
  </span>
  <div
    className="stat-value"
    data-radium={true}
    style={
      Object {
        "color": "#FFFFFF",
        "fontSize": "16px",
        "fontWeight": "bold",
        "height": "19px",
        "lineHeight": "19px",
        "textAlign": "center",
      }
    }
  >
    <div
      id="loadingContainer"
    >
      <IconSVG
        id="loadingIcon"
        name="loadingWhite"
        style={
          Object {
            "height": "15px",
            "position": "relative",
            "top": "-5px",
            "width": "15px",
          }
        }
      />
    </div>
  </div>
  <div
    className="agent-stat"
    data-radium={true}
    style={
      Object {
        "color": "#FFFFFF",
        "fontSize": "12px",
        "fontWeight": "600",
        "height": "10px",
        "lineHeight": "10px",
        "overflow": "hidden",
        "padding": "0 4px",
        "textAlign": "center",
        "textOverflow": "ellipsis",
        "whiteSpace": "nowrap",
        "width": "100%",
      }
    }
  >
    mockName
  </div>
</div>
`;
